sdk/compiler/damlc/lib/DA/Cli/Damlc/Command/MultiIde/Prefixing.hs

```diff
@@ -20,100 +21,102 @@ module DA.Cli.Damlc.Command.MultiIde.Prefixing (
+
+progressTokenPrefixFromMaybe :: Maybe T.Text -> ProgressTokenPrefix
+progressTokenPrefixFromMaybe = maybe ClientPrefix SubIDEPrefix
+
+-- | Reversible ProgressToken prefixing
+-- Given ProgressTokens can be int or text, we encode them as text as well as a tag to say if the original was an int
+-- Such that ProgressNumericToken 10   -> ProgressTextToken "iPREFIX-10"
+--       and ProgressTextToken "hello" -> ProgressTextToken "tPREFIX-hello"
+addProgressTokenPrefix :: ProgressTokenPrefix -> LSP.ProgressToken -> LSP.ProgressToken
+addProgressTokenPrefix NoPrefix t = t
+addProgressTokenPrefix prefix (LSP.ProgressNumericToken t) = LSP.ProgressTextToken $ "i" <> progressTokenPrefixToText prefix <> "-" <> T.pack (show t)
+addProgressTokenPrefix prefix (LSP.ProgressTextToken t) = LSP.ProgressTextToken $ "t" <> progressTokenPrefixToText prefix <> "-" <> t
+
+progressTokenSplitPrefix :: T.Text -> (T.Text, Maybe T.Text)
+progressTokenSplitPrefix = bimap T.tail (mfilter (/="client") . Just) . swap . T.breakOn "-"
+
+-- Removes prefix, returns the subIDE prefix if the token was created by a subIDE
+removeProgressTokenPrefix :: LSP.ProgressToken -> (LSP.ProgressToken, ProgressTokenPrefix)
+removeProgressTokenPrefix (LSP.ProgressTextToken (T.uncons -> Just ('i', rest))) =
+  bimap (LSP.ProgressNumericToken . read . T.unpack) progressTokenPrefixFromMaybe $ progressTokenSplitPrefix rest
+removeProgressTokenPrefix (LSP.ProgressTextToken (T.uncons -> Just ('t', rest))) =
+  bimap LSP.ProgressTextToken progressTokenPrefixFromMaybe $ progressTokenSplitPrefix rest
+removeProgressTokenPrefix t = (t, NoPrefix)
+
+-- Prefixes the SWindowWorkDoneProgressCreate and SProgress messages from subIDE. Rest are unchanged.
+addProgressTokenPrefixToServerMessage :: T.Text -> LSP.FromServerMessage -> LSP.FromServerMessage
+addProgressTokenPrefixToServerMessage prefix (LSP.FromServerMess LSP.SWindowWorkDoneProgressCreate req) =
+  let prefixedToken = addProgressTokenPrefix (SubIDEPrefix prefix) $ req ^. LSP.params . LSP.token
+   in LSP.FromServerMess LSP.SWindowWorkDoneProgressCreate $ req & LSP.params . LSP.token .~ prefixedToken
+addProgressTokenPrefixToServerMessage prefix (LSP.FromServerMess LSP.SProgress notif) =
+  case removeProgressTokenPrefix $ notif ^. LSP.params . LSP.token of
+    -- ProgressToken was created by this subIDE, add its usual prefix
+    (unprefixedToken, NoPrefix) ->
+      let prefixedToken = addProgressTokenPrefix (SubIDEPrefix prefix) unprefixedToken
+       in LSP.FromServerMess LSP.SProgress $ notif & LSP.params . LSP.token .~ prefixedToken
+    -- ProgressToken was created by client, send back the unprefixed token
+    (unprefixedToken, ClientPrefix) -> LSP.FromServerMess LSP.SProgress $ notif & LSP.params . LSP.token .~ unprefixedToken
+    (_, SubIDEPrefix t) -> error $ "SubIDE with prefix " <> T.unpack t <> " is somehow aware of its own prefixing. Something is very wrong."
```

Comment:
Is this condition so severe that we want to halt the IDE over it, or is it something we could warn and ignore for instead?

Reply:
ditto for "ignore-erroneous-subide"?